oauth2_http/java/com/google/auth/oauth2/S2A.java

```diff
@@ -0,0 +1,110 @@
+package com.google.auth.oauth2;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.json.JsonObjectParser;
+import com.google.api.client.util.GenericData;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ServiceLoader;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Utilities to fetch the S2A (Secure Session Agent) address from the mTLS configuration.
+ *
+ * <p>mTLS configuration is queried from the MDS MTLS Autoconfiguration endpoint.
+ */
+@ThreadSafe
+public final class S2A {
+  public static final String S2A_CONFIG_ENDPOINT_POSTFIX =
+      "/computeMetadata/v1/instance/platform-security/auto-mtls-configuration";
+
+  public static final String METADATA_FLAVOR = "Metadata-Flavor";
+  public static final String GOOGLE = "Google";
```

Comment:
Do these constants need to be public? Can they be package-private? Or are they intended to be used by users?

Reply:
They aren't intended for users, but they are used in the tests (see: MockMetadataServerTransport.java). In this case would you prefer we make these private and add static getters so tests can use them?

Reply:
Can these package-private? I think it should be accessible

Reply:
Thanks for pointing that out! Done in https://github.com/googleapis/google-auth-library-java/pull/1400/commits/ae545c82e217353bf086a856674ac7ae54e59e35